include/pipe.h

```diff
@@ -53,13 +53,41 @@ class Pipe {
     darcy_friction_ = darcy_friction;
   }
 
+  //! Assign pipe roughness coefficient
+  //! \param[in] pipe_roughness roughness coefficient of the pipe
+  void pipe_roughness(double pipe_roughness) {
+    pipe_roughness_ = pipe_roughness;
+  }
+
   //! Calculate and return discharge using Darcy-Weisbach equation:
-  //! dhead = (8*darcy_factor*pow(discharge,2)/(pow(M_PI,2)*g*pow(2*radius,5))
+  //! dhead/length =
+  //! (8*darcy_factor*pow(discharge,2)/(pow(M_PI,2)*g*pow(2*radius,5)) That is,
+  //! discharge =
+  //! sqrt(dhead/length*pow(M_PI,2)*g*pow(2*radius,5)/(8*darcy_friction));
+  //! \retval discharge_ discharge in the pipe
+  double discharge_dw();
+
+  //! Calculate and return discharge using Hazen-Williams equation:
+  //! dhead/length =
+  //! (10.67*pow(discharge,1.852)/(pow(pipe_roughness,1.852)*pow(2*radius,4.8704))
   //! That is, discharge =
-  //! sqrt(dhead*pow(M_PI,2)*g*pow(2*radius,5)/(8*darcy_friction));
+  //! pow((dhead*pow(pipe_roughness,1.852)*pow(2*radius,4.8704)/(10.67*length)),1/1.852);
   //! \retval discharge_ discharge in the pipe
-  double discharge();
+  double discharge_hw();
 
+  //! Calculate and return head loss over the pipe using Darcy-Weisbach
+  //! equation: dhead =
+  //! length*(8*darcy_factor*pow(discharge,2)/(pow(M_PI,2)*g*pow(2*radius,5))
+  //! \retval headloss_  headloss over the pipe
+  double headloss_dw();
+
+  //! Calculate and return headloss over the pipe using Hazen-Williams equation:
+  //! dhead/length
+  //! = 10.67*length*pow(discharge,1.852)/(pow(pipe_roughness,1.852)*pow(2*radius,4.8704))
+  //! \retval headloss_ headloss over the pipe
+  double headloss_hw();
+
+  //! Assign maximum allowable velocity
   //! Assign maximum allowable velocity
   //! \param[in] max_flowrate maximum allowable velocity of flow in the pipe
   void max_velocity(double max_velocity) { max_velocity_ = max_velocity; }
```

Comment:
Can we have initilalise flow conditions instead of getters and setters for every flow variable?